internal/controller/deployableartifact/controller.go

```diff
@@ -50,17 +51,64 @@ type Reconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.1/pkg/reconcile
 func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	// Fetch the DeployableArtifact instance
+	deployableartifact := &choreov1.DeployableArtifact{}
+	if err := r.Get(ctx, req.NamespacedName, deployableartifact); err != nil {
+		if apierrors.IsNotFound(err) {
+			// The DeployableArtifact resource may have been deleted since it triggered the reconcile
+			logger.Info("DeployableArtifact resource not found. Ignoring since it must be deleted.")
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object
+		logger.Error(err, "Failed to get DeployableArtifact")
+		return ctrl.Result{}, err
+	}
+
+	// Keep a copy of the original object for comparison
```

Comment:
```suggestion
        func shouldIgnoreReconcile(deployableArtifact *choreov1.DeployableArtifact) bool {
	        return meta.FindStatusCondition(deployableArtifact.Status.Conditions, string(controller.TypeAvailable)) != nil 
        }
	// Keep a copy of the original object for comparison
```

func shouldIgnoreReconcile(deployableArtifact *choreov1.DeployableArtifact) bool {
	return meta.FindStatusCondition(deployableArtifact.Status.Conditions, string(controller.TypeAvailable)) != nil
	}

Reply:
fixed